File: Android/app/src/main/java/protocol/MessagePacket.kt
package com.offlinemessenger.android.protocol

import java.util.UUID


class MessagePacket {

    var type: MessageType = MessageType.Chat

    var sessionId: UUID = UUID.randomUUID()

    var messageId: UUID = UUID.randomUUID()

    var timestamp: Long = 0

    var payload: ByteArray = ByteArray(0)

    var nonce: ByteArray = ByteArray(0)

    var tag: ByteArray = ByteArray(0)
}

File: Windows/OfflineMessenger.Core/Protocol/MessagePacket.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OfflineMessenger.Core.Protocol;

public class MessagePacket
{
    public MessageType Type { get; set; }
    public Guid SessionId { get; set; }
    public Guid MessageId { get; set; }
    public long Timestamp { get; set; }

    public byte[] Payload { get; set; }

    public MessageStatus Status { get; set; }

    public MessageDirection Direction { get; set; }

    // 🔐 ДОБАВЛЯЕМ
    public byte[] Nonce { get; set; }
    public byte[] Tag { get; set; }
}